### PR TITLE
Add warnings for collision with builtin variable (#1637)

### LINKED ISSUE #1637: Taipy GUI cannot use variables that have a name that is a builtin function
### What went wrong? 🤔

If a variable has the same name as a builtin function, then Taipy GUI retrieves this builtin instead of the variable value when resolving the binding.

### Expected Behavior

One should be able not to worry about variable names.

### Steps to Reproduce Issue

This code:
```python
from taipy.gui import Gui

max = 12345

Gui(page="<|{max}|text|>").run()
```
fails as 'max' in a Python builtin function.

### Additional Context

```bash
The error would be:
`Object of type builtin_function_or_method is not JSON serializable (value: <built-in function <name>>)`
```

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Resolve #1637

### CODE REVIEW

File: taipy/gui/utils/_evaluator.py [hunk 1]
```diff
@@ -106,7 +107,9 @@ def _analyze_expression(
             for node in ast.walk(st):
                 if isinstance(node, ast.Name):
                     var_name = node.id.split(sep=".")[0]
-                    if var_name not in args and var_name not in targets and var_name not in non_vars:
+                    if var_name in builtin_vars:
+                        _warn(f"Variable '{var_name}' can't be used in taipy expression as it collides with python built-in variable")
```
Comment (FabienLelaquais) [inline taipy/gui/utils/_evaluator.py]:
```suggestion
                        _warn(f"Variable '{var_name}' cannot be used in Taipy expressions as its name collides with a Python built-in identifier.")
```

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
Nice job!
Just reworded the warning message.

**Comment by dinhlongviolin1** [comment]
@FabienLelaquais Thanks for the rewording. Looks really good!

**Review by namnguyen20999 — APPROVED** [review]

**Review by FabienLelaquais — APPROVED** [review]
👍👍👍

**Pull request merged**